client/lib/sites-list/list.js

```diff
@@ -394,7 +394,7 @@ SitesList.prototype.setRecentlySelectedSite = function( siteID ) {
 };
 
 SitesList.prototype.getRecentlySelected = function() {
-	if ( ! this.recentlySelected.length ) {
+	if ( ! this.recentlySelected.length || ! this.initialized ) {
```

Comment:
One thing that occurred to me, you don't need the check for `! this.recentlySelected.length` since you set it to an empty array by default and would then return an empty array here.